apps/admin/src/app/(dashboard)/hackers/page.tsx

```diff
@@ -382,3 +386,206 @@ export function HackathonsTable() {
 
   return <DataTable columns={hackerColumns} data={hackers} />;
 }
+
+export function ConfirmTable({ updateCounts }: TableProps) {
+  const { data: hackers, isPending, error } = api.hacker.adminAll.useQuery();
+
+  const updateHacker = api.hacker.adminUpdate.useMutation({
+    onSuccess: () => {
+      toast("Success!", {
+        description: "Hacker Status Updated.",
+      });
+      // Refetch hackers data after updating
+    },
+    onError: (error) => {
+      toast("Error!", {
+        description: error.message,
+      });
+    },
+  });
+
+  const [allHackers, setAllHackers] = useState<typeof hackers>([]);
+  const [selectedHacker, setSelectedHacker] = useState(
+    hackers ? hackers[0] : null,
+  );
+
+  useEffect(() => {
+    const filteredHackers = hackers?.filter(
+      (hacker) => hacker.status === "confirmed",
+    );
+    setAllHackers(filteredHackers);
+    if (filteredHackers) {
+      if (filteredHackers.length > 0) {
+        setSelectedHacker(filteredHackers[0]);
+      } else {
+        setSelectedHacker(null);
+      }
+    }
+  }, [hackers]);
+
+  if (isPending) return <div>Loading...</div>;
+  if (error) return <div>Error: {error.message}</div>;
+
+  const confirmedCount = allHackers?.length;
+
+  const moveToNextApplicant = () => {
+    const currentIndex = allHackers?.findIndex(
+      (h) => h.id === selectedHacker?.id,
+    );
+    if (
+      allHackers &&
+      currentIndex &&
+      currentIndex > -1 &&
+      currentIndex < allHackers.length - 1
+    ) {
+      setSelectedHacker(allHackers[currentIndex + 1]);
+    } else {
+      setSelectedHacker(null); // No more applicants
+    }
+  };
+
+  type Status =
```

Comment:
Shouldn't we import this from validators?

Reply:
ur so right king

Reply:
See changes (it comes from consts, not validators)